tests/test_compare_dictconfig_vs_dict.py

```diff
@@ -123,6 +120,14 @@ def test__delitem__(
                 del cfg[key]
                 assert key not in cfg
 
+    def test__delitem__struct_mode(
+        self, python_dict: Any, cfg: DictConfig, key: Any, struct_mode: bool
+    ) -> None:
+        """Ensure that __delitem__ fails in struct_mode"""
+        if struct_mode:
+            with raises(ConfigTypeError):
+                del cfg[key]
+
```

Comment:
if this test only operates when struct_mode is True, why even pass it?
Why not this?
```suggestion
    def test__delitem__struct_mode(
        self, python_dict: Any, cfg: DictConfig, key: Any
    ) -> None:
        """Ensure that __delitem__ fails in struct_mode"""
        OmegaConf.set_struct(cfg, True)    
        with raises(ConfigTypeError):
            del cfg[key]

```

Reply:
The reason is that the `cfg` fixture is parametrized by `struct_mode`; by passing in `cfg` in the test function signature, the test gets called once with `struct_mode == True` and once with `struct_mode == False`.

I guess a clearer setup could be to have `cfg_struct_mode` and `cfg_no_struct_mode` fixtures in addition to the `cfg` fixture.

Reply:
or cfg and cfg_struct:
```
cfg - just a default DictConfig
cfg_struct:
  cfg with struct False
  cfg with struct True
```